### PR TITLE
Convert tags, skip_tags, recognized_tags to sets; fix doctests; f-strings

### PR DESCRIPTION
**Pull request by willkg**
This converts the `tags` argument to `BleachHTMLParser` to be a set.

This converts the `skip_tags` and `recognized_tags` to linkify things to be sets.

This updates the documentation fixing example code so that `tags`, `skip_tags`, and `recognized_tags` are all sets.

This also converts some string interpolation from `%s` style to f-strings.

### CODE REVIEW

**Comment by willkg** [comment]
I need to go through this again and double-check that we've got the docs, tests, and everything correct.

File: bleach/linkifier.py [hunk 1]
```diff
@@ -120,17 +120,18 @@ def __init__(
...
+            want linkify to skip any tags
 
         :arg bool parse_email: whether or not to linkify email addresses
 
         :arg url_re: url matching regex
 
         :arg email_re: email matching regex
 
-        :arg list recognized_tags: the list of tags that linkify knows about;
+        :arg set recognized_tags: the list of tags that linkify knows about;
```
Comment (willkg) [inline bleach/linkifier.py:134]: Oops--missed this "list".

File: docs/clean.rst [hunk 2]
```diff
@@ -224,9 +228,10 @@ This adds smb to the Bleach-specified set of allowed protocols:
 
    >>> import bleach
 
+   >>> my_protocols = bleach.ALLOWED_PROTOCOLS.union({'smb'})
```
Comment (willkg) [inline docs/clean.rst:231]: Switching these to sets makes them slightly more awkward to manipulate. :(

**Comment by willkg** [comment]
@g-k This is the last code change before the next release. Can you look it over?

I verified the tests are passing in the right stuff by adding:

```
assert isinstance(tags, (type(None),  set, frozenset)), f"wrong type {type(tags) = }"
```

and then running the tests and building the docs (which runs the doctests). That sussed out a bunch of issues. I did the same thing for `skipt_tags`, `recognized_tags`, and `protocols`.

File: docs/clean.rst (hunk 2 above)
Comment (willkg) [reply docs/clean.rst]: I forgot we can use `|`. That makes this better.

**Comment by willkg** [comment]
As a side note, while fiddling with sets and f-strings, I wondered if we'd get any performance gain.

I use the Standup data (remember Standup?) to test bleach.clean since it's a pretty straight-forward data set of messages with tags and urls and a bunch of stuff in it.

I used Python 3.10.9.

* 5.0.0: bleach.clean on 58,630 items 10x: minimum 2.793s
* with this PR: bleach.clean on 58,630 items 10x: minimum 2.304s

Seems like we got a small speedup with `bleach.clean`, so that's cool.

**Review by g-k — APPROVED** [review]
r+ lgtm and it's nice to get perf and readability wins

is the Standup data from the app people used to post their (stand up) status?

**Comment by willkg** [comment]
Yeah! pmac and I ran [Standup](https://github.com/mozilla/standup) which was the last project I worked on that used Bleach. I have a data file of the message strings.

I'll do another pass on this tomorrow and then land it and finish up the release.

Thank you!

**Pull request merged**